buildSrc/src/main/java/org/elasticsearch/gradle/precommit/JarHellPrecommitPlugin.java

```diff
@@ -20,12 +19,6 @@
     @Override
     public TaskProvider<? extends Task> createTask(Project project) {
         Configuration jarHellConfig = project.getConfigurations().create("jarHell");
-        if (BuildParams.isInternal() && project.getPath().equals(":libs:elasticsearch-core") == false) {
```

Comment:
We moved setting the dependencies here to `InternalPrecommitTask`. to not rely on internal BuildParams and keep this public Plugin cleaner with none internal specific polluted code. We could alternatively have an InternalJarHellPrecommitPlugin but as we want to revisit default dependency declarations in our plugins and our internal build that's likely too much.